clarify test config values (#3209)
cypress-io/cypress-documentation: source/_partial/allowed_test_config.md

- `animationDistanceThreshold`
- `baseUrl`

> - `browser` **note:** this is the only option that allows to skip a test or a suite of tests depending on the current browser

Reviewer: ```suggestion
- `browser` **note:** filters whether the tests or a suite of tests runs depending on the current browser
```

- `defaultCommandTimeout`
- `execTimeout`

> - `env` **note:** Cypress merges the object of environment variables

Reviewer: ```suggestion
- `env` **note:** Provided environment variables will be merged with current environment variables.
```
